scripts/consume_sitemaps.py

```diff
@@ -170,62 +171,52 @@ def fetch_sitemap_files(authority, last_updated_date, thing_ids: typing.Dict[str
-                    things_fetcher = thing_fut.result()
-                    if things_fetcher is not None and things_fetcher.json_things is not None:
-                        global __NUM_THINGS_FETCHED
-                        __NUM_THINGS_FETCHED += len(things_fetcher.json_things)
-                        logging.info(f"About to process {len(things_fetcher.json_things)} things")
-                        for json_thing in things_fetcher.json_things:
-                            json_thing["tstamp"] = datetime.datetime.now()
-                            identifiers = thing_identifiers_from_resolved_content(authority, json_thing["resolved_content"])
-                            identifiers.append(json_thing["id"])
-                            json_thing["identifiers"] = json.dumps(identifiers)
-                            # remove the pk as that isn't guaranteed to be the same
-                            del json_thing["primary_key"]
-                            if thing_ids.__contains__(json_thing["id"]):
-                                # existing row in the db, for the update to work we need to insert the pk into the dict
-                                json_thing["primary_key"] = thing_ids[json_thing["id"]]
-                                current_existing_things_batch.append(json_thing)
-                            else:
-                                current_new_things_batch.append(json_thing)
-                        db_session.bulk_insert_mappings(
-                            mapper=Thing, mappings=current_new_things_batch, return_defaults=False
-                        )
-                        db_session.bulk_update_mappings(
-                            mapper=Thing, mappings=current_existing_things_batch
-                        )
-                        db_session.commit()
-                        logging.info(f"Just processed {len(things_fetcher.json_things)} things")
-                    else:
-                        logging.error(f"Error fetching thing for {things_fetcher.url}")
-                    thing_futures.remove(thing_fut)
-                    if len(thing_futures) < REFETCH_LENGTH:
-                        # if we are running low on things to process, kick off the next batch to download
-                        fetched_all_things_for_current_sitemap_file = (
-                            construct_thing_futures(
-                                thing_futures,
-                                sitemap_file_iterator,
-                                rsession,
-                                thing_executor,
-                            )
-                        )
+            # Then read out results and save to the database after the queue is filled to capacity.
+            # Provided there are more urls in the iterator, return to the top of the loop to fill the queue again
+            for thing_fut in concurrent.futures.as_completed(thing_futures):
```

Comment:
Previously, we were appending to the `thing_futures` list while we were servicing this loop.  I think I had structure it that way to avoid overwhelming the server with a bunch of requests at once.  That was the wrong approach -- limiting the max_workers to 1 allows one request to the server at a time while we are concurrently processing results and writing to the database.